StorageUtility/Windows/Device.cpp

```diff
@@ -0,0 +1,43 @@
+/*
+<License>
+Copyright 2015 Virtium Technology
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http ://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+</License>
+*/
+#include "Device.h"
+#include "DeviceData.h"
+
+namespace vtStor
+{
+
+cDevice::cDevice(PSP_DEVINFO_DATA DevInfoData, PSP_DEVICE_INTERFACE_DATA DevInterfaceData, PSP_INTERFACE_DEVICE_DETAIL_DATA PDevDetailData)
+{
+    m_Data.clear();
+    m_Data.insert({ SP_DEVINFO_DATA_TYPE, cDeviceData<PSP_DEVINFO_DATA>::ToVoidPointer(DevInfoData) });
+    m_Data.insert({ SP_DEVICE_INTERFACE_DATA_TYPE, cDeviceData<PSP_DEVICE_INTERFACE_DATA>::ToVoidPointer(DevInterfaceData) });
+    m_Data.insert({ PSP_INTERFACE_DEVICE_DETAIL_DATA_TYPE, cDeviceData<PSP_INTERFACE_DEVICE_DETAIL_DATA>::ToVoidPointer(PDevDetailData) });
```

Comment:
Exactly, it causes runtime error
Two pointers point to a same location and when one pointer deallocated then the other will point to a deallocated location
That's because the "PSP_INTERFACE_DEVICE_DETAIL_DATA pDevDetailData" called LocalFree after function EnumerateDevices done
I am not sure to make a good solution for this (maybe I should use a deep copy, clone or something)
Could you please give me the suggestion to solve this?